src/cryptonote_core/service_node_deregister.cpp

```diff
@@ -0,0 +1,734 @@
+      "4f7d926bfc5f9a264a3a0972d09087f38eaa76b2879e61715d01017194386a72",
+      "3ae858e31c725bdc414984b3bd51c811a8cc3be60cc5d5825233d40afd8ece43",
+      "f1601d0e7000065e0a23a25b83d0115c44749deff8037241caaa2f2677bf24c5",
+      "851f450921e37545a804c63bc10d4d403eacc0aec823ad861c5233e84aa5e192",
+      "01494ec0e954c527ebb3dced7d8e0c1a2ef95b1183034dfdb70c8345d507490e",
+      "592010c8de072ac789e792b63e50b9048aecb64b0edbc8a7e0bb41500e10cf3f",
+      "d8e9769b34373e7eb2f82b91dc5046714595d8f80d1f8793ed0e84798c527ef6",
+      "095237eeb750ef8c0c10c96e89a54cfa448cac8f031ee2007700e5b11becaf3b",
+      "d685543dc209da64006fe922e1091b4b683ab6f05e6695133ad75f7762df1d17",
+      "c993d744c77580361968411ae1201ea0514a002825d7aa0af2d9428ef621e6e8",
+      "41c4948730283f4bfa498819444c03cf8eaa159e9fb8ad02a7364c31dd182279",
+      "4ed4b99af67859e051f49bb4dfb08c2d1b60993ad64752d326688db191dd7eab",
+      "6081920bd10a4dfc3eb8dd29e8fa7791f450a8081b6659af335ef87da30779aa",
+      "d66ac6f8dfed95688c1dab37129fcf6af793ec694d2afbabd4b488b8e5248d46",
+      "23acc0688ab2c620b5bab815f4685ccf7b39c09fec52aea539a6c29f794b1bde",
+      "fc1a3b72e819e5af4e64663ea959efd6dd91a1e803aaa80eecbc0129ac44728b",
+      "98c5fc86b4700ac8280500707e9d368b2330ddfe75329d915c55ce628abd942e",
+      "cf7fdb83942a2f4f4ed3acdc8a3b0a67b9561b8d16cde61ae5f821150498a537",
+      "23d0a95a178863b2c70685ef01e335452dda09e541aa5415830d73588ca2d702",
+      "bf4b8963a41856fd1629a9920bf8c47a177d874b88e957651c5e31e6b7dab3b4",
+      "8f487cc93154b36d423cd74db955320d22956fec83d484a68747fcc83682f7c3",
+      "6c43a23dc0d0b0496d4fc35434d4ae8ccb26e5d444ac1a1187fac1062f3bad6f",
+      "47138315af577191066a762eb34789d93266b9f2b2bede649c49fa13d6806bc6",
+      "f87fe98e826ea34eee3ff755178cb0bb8fa8d2775d51d761f1681253226e6720",
+    };
+
+    void init()
+    {
+      if (secret_view_keys.size() > 0) return;
+
+#define ARRAY_COUNT(array) (sizeof(array) / sizeof(array[0]))
+      secret_view_keys.resize(ARRAY_COUNT(secret_view_keys_str));
+      public_view_keys.resize(ARRAY_COUNT(secret_view_keys_str));
+      secret_spend_keys.resize(ARRAY_COUNT(secret_view_keys_str));
+      public_spend_keys.resize(ARRAY_COUNT(secret_view_keys_str));
+
+      for (size_t i = 0; i < ARRAY_COUNT(secret_view_keys_str); i++)
+      {
+        assert(epee::string_tools::hex_to_pod(secret_view_keys_str[i] , secret_view_keys[i]));
+        assert(epee::string_tools::hex_to_pod(secret_spend_keys_str[i], secret_spend_keys[i]));
+        assert(epee::string_tools::hex_to_pod(public_view_keys_str[i] , public_view_keys[i]));
+        assert(epee::string_tools::hex_to_pod(public_spend_keys_str[i], public_spend_keys[i]));
+      }
+#undef ARRAY_COUNT
+    }
+
+  };
+
+  struct copy_region
+  {
+    void const *ptr;
+    size_t num_bytes;
+  };
+
+  static crypto::hash hash_region_to_buf(void *buf, size_t buf_size, copy_region const *regions, int num_regions)
+  {
```

Comment:
Hey I know. This is super unnecessary. After refactoring I forgot to merge this function down to just make_hash_from.

Reply:
Fixed.